### PR TITLE
Improve documentation, especially on SNMP

### PR DESCRIPTION
**Pull request by pyhys**

### CODE REVIEW

File: doc/implementation_details.rst [hunk 1]
```diff
@@ -602,3 +610,19 @@ Run clang-format on staged files before committing::
...
+
+Workflow
+--------
+We have chosen to host the code on Github to ease the collaboration between
+users and different developers, and we take advantage of the standard
+Github workflow:
+
+* Open a "issue" for each separate bug found.
+* The developer forks the repository to her own account on Github, and
+  makes a local clone her the work station.
```
Comment (fredrikdanielmoller) [inline doc/implementation_details.rst:622]: Reformulate

**Review by fredrikdanielmoller — CHANGES_REQUESTED** [review]

File: doc/implementation_details.rst (hunk 1 above)
Comment (pyhys) [reply doc/implementation_details.rst]: Thanks, Done.

**Review by fredrikdanielmoller — APPROVED** [review]

File: CMakeLists.txt [hunk 2]
```diff
@@ -54,7 +54,7 @@ set(PNET_MAX_CR                 2
 set(PNET_MAX_MODULES            5
   CACHE STRING "Per API. Should be > 1 to allow at least one I/O module")
 set(PNET_MAX_SUBMODULES         3
-  CACHE STRING "Per module (3 needed for DAP)")
+  CACHE STRING "Per slot (3 needed for DAP)")
```
Comment (olbjo) [inline CMakeLists.txt:57]: Maybe reword string for MAX_MODULES. "Max slots per API" or "Simultaneously plugged modules per API. " The relationship between slot/module subslot/submodule is not really clear.

**Review by olbjo — APPROVED** [review]

File: CMakeLists.txt (hunk 2 above)
Comment (pyhys) [reply CMakeLists.txt:57]: I plan to rename MAX_MODULES to MAX_SLOTS in a separate PR (after your  stuff is merged).

**Review by fredrikdanielmoller — APPROVED** [review]

File: doc/implementation_details.rst [hunk 3]
```diff
@@ -602,3 +614,21 @@ Run clang-format on staged files before committing::
...
+--------
+We have chosen to host the code on Github to ease the collaboration between
+users and different developers, and we take advantage of the standard
+Github workflow:
+
+* Open a Github issue on https://github.com/rtlabs-com/p-net/ for each
+  separate bug found.
+* Fork the repository to your own account on Github, and make a local
+  clone on your work station.
+* Create a branch named ``githubNUMBER_description``, where ``NUMBER`` is
```
Comment (hefloryd) [inline doc/implementation_details.rst:629]: Will GH check the branch name to understand which issue it fixes? The usual method would be to mention the issue in the PR, and the submitter chooses whatever name they want for the branch. This workflow is probably already well documented on GH.

Comment (pyhys) [reply doc/implementation_details.rst:629]: Great, I will update the text.
